utils/appliance/__init__.py

```diff
@@ -2149,3 +2149,11 @@ def get_or_create_current_appliance():
 class CurrentAppliance(object):
     def __get__(self, instance, owner):
         return get_or_create_current_appliance()
+
+
+class Navigatable(object):
+
+    appliance = CurrentAppliance()
```

Comment:
the descriptor is no logner needed in that case

im starting to think we really want to take a look at the attrs lib, we are getting more and more mixings with attributes that have sane defaults